libraries/AP_AngleSensor/AP_AngleSensor.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_AngleSensor.h"
+
+#if AP_ANGLESENSOR_ENABLED
+
+#include "AP_AngleSensor_AS5048B.h"
+#include <AP_Logger/AP_Logger.h>
+#include <AP_Math/AP_Math.h>
+#include <GCS_MAVLink/GCS.h>
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_AngleSensor::var_info[] = {
+    // @Group: _
+    // @Path: AP_AngleSensor_Params.cpp
+    AP_SUBGROUPINFO(_params[0], "_", 0, AP_AngleSensor, AP_AngleSensor_Params),
```

Comment:
```suggestion
    // @Group: 1_
    // @Path: AP_AngleSensor_Params.cpp
    AP_SUBGROUPINFO(_params[0], "1_", 1, AP_AngleSensor, AP_AngleSensor_Params),
```
My preference would be to number the first instance too, just keep things more consistent. Also param indexes should start at 1.